File: CS/Sample/Models/Role.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.ComponentModel.DataAnnotations;

namespace Sample.Models
{
    public class Role
    {

         public Role() {
            RoleID = 0;
            RoleName = string.Empty;
        }

        public Role(int id, String name) {
            this.RoleID = id;
            this.RoleName = name;
        }

        [Key]
        public int RoleID { get; set; }

        [Required(ErrorMessage = "Role Name is required")]
        public string RoleName { get; set; }
    }
}

File: VB/Sample/Models/Role.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.ComponentModel.DataAnnotations

Namespace Sample.Models
    Public Class Role

         Public Sub New()
            RoleID = 0
            RoleName = String.Empty
         End Sub

        Public Sub New(ByVal id As Integer, ByVal name As String)
            Me.RoleID = id
            Me.RoleName = name
        End Sub

        <Key> _
        Public Property RoleID() As Integer

        <Required(ErrorMessage := "Role Name is required")> _
        Public Property RoleName() As String
    End Class
End Namespace